elytron/src/main/java/org/wildfly/extension/elytron/FileSystemRealmDefinition.java

```diff
@@ -229,6 +312,23 @@ public SecurityRealm get() throws StartException {
                             if (nameRewriter == null) {
                                 nameRewriter = NameRewriter.IDENTITY_REWRITER;
                             }
+                            KeyStore keyStore = keyStoreInjector.getOptionalValue();
+                            PrivateKey privateKey = null;
+                            PublicKey publicKey = null;
+                            if (keyStore != null) {
+                                    try {
+                                        keyStoreService = getModifiableKeyStoreService(keyStoreServiceRegistry, keyStoreName);
+                                        char[] keyPassword = getKeyStorePassword((KeyStoreService) keyStoreService);
+                                        privateKey = (PrivateKey) keyStore.getKey(keyPairAlias, keyPassword);
+                                        publicKey = keyStore.getCertificate(keyPairAlias).getPublicKey();
+                                        if (privateKey == null ^ publicKey == null) {
+                                           throw ROOT_LOGGER.invalidKeyPair(keyStoreName, keyPairAlias);
```

Comment:
It might also be good to have different exception messages depending on whether the private key or public key is null. As an example, you could take a look at what was done here:

https://github.com/wildfly/wildfly-core/blob/main/elytron/src/main/java/org/wildfly/extension/elytron/AdvancedModifiableKeyStoreDecorator.java#L345-L364